test/rhsmlib_test/test_collector.py

```diff
@@ -43,3 +44,16 @@ def test_get_arch(self):
     def test_get_platform_specific_info_provider(self):
         info_provider = firmware_info.get_firmware_collector(arch=platform.machine())
         self.assertTrue(info_provider is not None)
+
+
+class GetNonDmiUuid(unittest.TestCase):
+    def test_get_aarch64_firmware_collector(self):
+        firmware_provider_class = firmware_info.get_firmware_collector(arch='aarch64')
+        self.assertTrue(isinstance(firmware_provider_class, UuidFirmwareInfoCollector))
+
+    @mock.patch(OPEN_FUNCTION, mock.mock_open(read_data="356B6CCC-30C4-11B2-A85C-BBB0CCD29F36"))
+    def test_get_aarch64_uuid_collection(self):
+        firmware_provider_class = firmware_info.get_firmware_collector(arch='aarch64')
+        firmware_provider_class.arch = 'aarch64'
+        result = firmware_provider_class.get_all()
+        self.assertTrue(result['dmi.system.uuid'] == '356B6CCC-30C4-11B2-A85C-BBB0CCD29F36')
```

Comment:
BTW: I can see that the UUID is also visible as `virt.uuid`. So I would add one more assertion:

```python
        self.assertTrue(result['virt.uuid'] == '356B6CCC-30C4-11B2-A85C-BBB0CCD29F36')
```